tests/Tests/500__jupyterhub/jupyterhub-user-access.robot

```diff
@@ -0,0 +1,89 @@
+    Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    Launch JupyterHub From RHODS Dashboard Dropdown
+    Login Verify Regular User Allowed  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+
+
+*** Keywords ***
+Special User Testing Suite Setup
+  Set Library Search Order  SeleniumLibrary
+
+Login Verify Not Allowed
+    [Arguments]  ${username}  ${password}  ${auth}
+    Run Keyword and Ignore Error  Delete Group  rhods-admins
+    Run Keyword and Ignore Error  Delete Group  rhods-users
+    Create Group  tests/Tests/500__jupyterhub/test-data/rhods-users-empty.yaml
+    Login To Jupyterhub  ${username}  ${password}  ${auth}
+    ${authorization_required} =  Is Service Account Authorization Required
+    Run Keyword If  ${authorization_required}  Authorize jupyterhub service Account
+    User Is Not Allowed
+    Capture Page Screenshot  not-allowed-user-login-{index}.png
+    Delete Group  rhods-users
+
+Login Verify Admin Allowed
+    [Arguments]  ${username}  ${password}  ${auth}
+    Run Keyword and Ignore Error  Delete Group  rhods-admins
+    Copy File  tests/Tests/500__jupyterhub/test-data/rhods-admins-group.yaml  tests/Tests/500__jupyterhub/test-data/rhods-admins-group-modified.yaml
+    Run  sed -i 's/<user>/${username}/g' tests/Tests/500__jupyterhub/test-data/rhods-admins-group-modified.yaml
+    Create Group  tests/Tests/500__jupyterhub/test-data/rhods-admins-group-modified.yaml
+    Remove File  tests/Tests/500__jupyterhub/test-data/rhods-admins-group-modified.yaml
+    Login To Jupyterhub  ${username}  ${password}  ${auth}
+    ${authorization_required} =  Is Service Account Authorization Required
+    Run Keyword If  ${authorization_required}  Authorize jupyterhub service Account
+    User Is JupyterHub Admin
+    Capture Page Screenshot  not-allowed-user-login-{index}.png
+    Delete Group  rhods-admins
+
+Login Verify Regular User Allowed
+    [Arguments]  ${username}  ${password}  ${auth}
+    Run Keyword and Ignore Error  Delete Group  rhods-users
+    Copy File  tests/Tests/500__jupyterhub/test-data/rhods-users-group.yaml  tests/Tests/500__jupyterhub/test-data/rhods-users-group-modified.yaml
+    Run  sed -i 's/<user>/${username}/g' tests/Tests/500__jupyterhub/test-data/rhods-users-group-modified.yaml
+    Create Group  tests/Tests/500__jupyterhub/test-data/rhods-users-group-modified.yaml
+    Remove File  tests/Tests/500__jupyterhub/test-data/rhods-users-group-modified.yaml
+    Login To Jupyterhub  ${username}  ${password}  ${auth}
+    ${authorization_required} =  Is Service Account Authorization Required
+    Run Keyword If  ${authorization_required}  Authorize jupyterhub service Account
+    User Is Not JupyterHub Admin
+    Capture Page Screenshot  not-allowed-user-login-{index}.png
+    Delete Group  rhods-users
```

Comment:
If using the ldap users, I think it would be feasible to have a single keyword for all three cases with a `type=user/admin/noaccess` argument and an `IF` block